shell.nix

```diff
@@ -5,13 +5,19 @@ pkgs.mkShell {
 
   packages = with pkgs; [
     cargo
+    rustc
     nodejs
     gcc
+    pnpm
+    git
 
```

Comment:
Why is this random blank line here? Also, is there a reason these packages aren't just sorted lexicographically?

Reply:
It's a common style in nix shells meant to distinguish "build inputs" (runtime dependencies like libraries) and "native build inputs" (compile time dependencies like compilers). Cmake (which was here before I added some newer required packages) should probably be moved up to the first section, (the "native build inputs") if we want to keep this style. This style overall is also not necessary since this isn't a nix derivation, so whatever style (with or without the blank line to separate build and native build inputs) is really fine

Reply:
As for the lexicographical order, they were just put wherever seemed fine. I'll update to sort them

Reply:
You can move the cmake dep.